rules/manager.go

```diff
@@ -956,16 +961,16 @@ func (m *Manager) LoadGroups(
 				))
 			}
 
-			groups[groupKey(rg.Name, fn)] = NewGroup(rg.Name, fn, itv, rules, shouldRestore, m.opts)
+			groups[groupKey(fn, rg.Name)] = NewGroup(rg.Name, fn, itv, rules, shouldRestore, m.opts)
 		}
 	}
 
 	return groups, nil
 }
 
 // Group names need not be unique across filenames.
-func groupKey(name, file string) string {
-	return name + ";" + file
+func groupKey(file, name string) string {
+	return file + ";" + name
```

Comment:
Won't this change metrics for users? Lets not change this lightly.

Reply:
It does not. Everything was inverted in the code. Some functions called it in the wrong order (for internal maps), this PR makes the internal map consistent with the metric.